salt/states/idem.py

```diff
@@ -129,8 +130,8 @@ def state(
         cache_dir=cache_dir or hub.OPT.idem.cache_dir,
         sls=SLSs,
         test=test,
-        acct_file=acct_file or hub.OPT.acct.acct_file,
-        acct_key=acct_key or hub.OPT.acct.acct_key,
+        acct_file=acct_file or os.environ.get("ACCT_FILE", hub.OPT.acct.acct_file),
```

Comment:
This seems like a weird place to be pulling in environment variables to override config options. Should this be handled by whatever populates hub.OPT?

Reply:
Ideally yes, but in salt/utils/idem.py I have to tell pop-config not to parse the CLI (because salt is using), which means hub.OPT also doesn't get environment variables

Reply:
It's something that could be fixed in pop-config and then made better here, I'll look into it :+1:

Reply:
yes, I think that is the right way to do this. We should be able to call hub.pop.config.load and pass in parse-cli=False, but that might avoid OS vars, so we might need to update pop-config to allow for cli to be avoided but osvars to be included